msal/application.py

```diff
@@ -230,32 +232,80 @@ def __init__(
-
-            *Added in version 1.13.0*:
-            It can also be a completely pre-signed assertion that you've assembled yourself.
-            Simply pass a container containing only the key "client_assertion", like this::
+            .. admonition:: Using ``public_certificate`` to support Subject Name/Issuer Auth
 
-                {
-                    "client_assertion": "...a JWT with claims aud, exp, iss, jti, nbf, and sub..."
-                }
+                *Added in version 0.5.0*:
+                public_certificate (optional) is public key certificate
+                which will be sent through 'x5c' JWT header only for
+                subject name and issuer authentication to support cert auto rolls.
+
+                Per `specs <https://tools.ietf.org/html/rfc7515#section-4.1.6>`_,
+                "the certificate containing
+                the public key corresponding to the key used to digitally sign the
+                JWS MUST be the first certificate.  This MAY be followed by
+                additional certificates, with each subsequent certificate being the
+                one used to certify the previous one."
+                However, your certificate's issuer may use a different order.
+                So, if your attempt ends up with an error AADSTS700027 -
+                "The provided signature value did not match the expected signature value",
+                you may try use only the leaf cert (in PEM/str format) instead.
+
+            .. admonition:: Supporting raw assertion obtained from elsewhere
+
+                *Added in version 1.13.0*:
+                It can also be a completely pre-signed assertion that you've assembled yourself.
+                Simply pass a container containing only the key "client_assertion", like this::
+
+                    {
+                        "client_assertion": "...a JWT with claims aud, exp, iss, jti, nbf, and sub..."
+                    }
+
+            .. admonition:: Supporting workload identity federated by Managed Identity
+
+                *Added in version 1.29.0*:
+                A confidential client app can authenticate via a managed identity.
+                This is known as "federated identity credential (FIC)" or
+                `"Workload identity federation" <https://learn.microsoft.com/entra/workload-id/workload-identity-federation>`_.
+
+                Once you setup the federation, the following declarative API
+                takes care of the managed identity token acquisition for you.
+                You just need to assign ``client_credential``
+                with an instance of :py:class:`msal.SystemAssignedManagedIdentity`
+                or :py:class:`msal.UserAssignedManagedIdentity`, for example::
+
+                    app = msal.ConfidentialClientApplication(
+                        "my_client_id",
+                        client_credential=msal.SystemAssignedManagedIdentity(),
+                        ...)
```

Comment:
My understanding is that **federated identity credential** can only be configured on **user-assigned managed identity**, not **system-assigned managed identity**, according to

https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation

> You use workload identity federation to configure a [user-assigned managed identity](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-manage-user-assigned-managed-identities) or [app registration](https://learn.microsoft.com/en-us/entra/identity-platform/app-objects-and-service-principals) in Microsoft Entra ID to trust tokens from an external identity provider (IdP), such as GitHub or Google.

https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-create-trust-user-assigned-managed-identity

> Configuring a federated identity credential on a system-assigned managed identity isn't supported.

Reply:
I see. It looks like you are creating federation between an app and a user-assigned managed identity.

Reply:
> I see. It looks like you are creating federation between an app and a user-assigned managed identity.

Yes. As described in the title of this PR, it is about a Confidential Client Application (CCA) federated by a Managed Identity (MI). It is NOT about letting a (User-assigned) MI federated by other IdP.

In particular, I tried a CCA federated by a system-assigned MI, and it worked.